src/pykeen/training/training_loop.py

```diff
@@ -673,3 +729,52 @@ def _free_graph_and_cache(self):
         self.model.regularizer.reset()
         # The cache of the previous run has to be freed to allow accurate memory availability estimates
         torch.cuda.empty_cache()
+
+    def save_state(self, path: str) -> None:
+        """Save the state of the training loop.
+
+        :param path:
+            Path of the file where to store the state in.
+        """
+        logger.debug("=> Saving checkpoint.")
+        torch.save(
+            {
+                'epoch': self._epoch,
+                'loss': self.losses_per_epochs,
+                'model_state_dict': self.model.state_dict(),
+                'optimizer_state_dict': self.optimizer.state_dict(),
+                'checksum': self.checksum,
+            },
+            path,
+        )
+        logger.info("=> Saved checkpoint.")
+
+    def load_state(self, path: str) -> int:
```

Comment:
How should this function get used? Would you please provide an example?

Reply:
I have made it a private function now, because imo a user should not fiddle around with these functions. The entire functionality is built to work automatically and consistently by providing a checkpoint file.